main.c

```diff
@@ -0,0 +1,121 @@
+
+#include <stdio.h>
+#include <stdlib.h>
+#include <string.h>
+#include <ctype.h>
+
+#define MAX_LEN 100
+
+// Function to check if the character is a valid digit or operator

```

Comment:
Let's not write unnecessary comments. Code should be self explanatory and it should not require comments.